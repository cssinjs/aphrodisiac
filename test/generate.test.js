import {create} from 'jss'
import aphroditeJss from '../src'

test('css returns a CSS string for a single property', () => {
  let aphrodite = aphroditeJss(create())
  let sheet = aphrodite.StyleSheet.create({
    red: {
      color: 'red'
    }
  })
  aphrodite.css(sheet.red)
  
  expect(aphrodite.toString()).toMatchSnapshot()
})

test('css implements override logic', () => {
  let aphrodite = aphroditeJss(create())
  let sheet = aphrodite.StyleSheet.create({
    red: {
      color: 'red'
    },
    blue: {
      color: 'blue'
    }
  })
  aphrodite.css(sheet.red, sheet.blue)
  
  expect(aphrodite.toString()).toMatchSnapshot()
})

test('css supports media queries', () => {
  let aphrodite = aphroditeJss(create())
  let sheet = aphrodite.StyleSheet.create({
    '@media (max-width: 400px)': {
      red: {
        color: 'red'
      }
    }
  })
  
  expect(aphrodite.toString()).toMatchSnapshot()
})

test('css supports keyframe', () => {
  let aphrodite = aphroditeJss(create())
  let styles = aphrodite.StyleSheet.create({
    '@keyframes anim1': {
      '0%': {
        transform: 'scale(0)'
      },
      '100%': {
        transform: 'scale(1)'
      }
    }
  })

  expect(aphrodite.toString()).toMatchSnapshot()
})

test('css supports font faces', () => {
  let aphrodite = aphroditeJss(create())
  let sheet = aphrodite.StyleSheet.create({
    '@font-face': {
      'font-family': 'Roboto',
      'src': "local('Roboto'), local('Roboto-Regular'), url('https://fonts.gstatic.com/s/roboto/v18/KFOmCnqEu92Fr1Mu72xKKTU1Kvnz.woff2') format('woff2')"
    }
  })
  
  expect(aphrodite.toString()).toMatchSnapshot()
})
