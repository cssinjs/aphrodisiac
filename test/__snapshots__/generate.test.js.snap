// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`css implements override logic 1`] = `
".red-3870469943--blue-268834934 {
  color: blue;
}"
`;

exports[`css returns a CSS string for a single property 1`] = `
".red-3870469943 {
  color: red;
}"
`;

exports[`css supports font faces 1`] = `
"@font-face {
  font-family: Roboto;
  src: local('Roboto'), local('Roboto-Regular'), url('https://fonts.gstatic.com/s/roboto/v18/KFOmCnqEu92Fr1Mu72xKKTU1Kvnz.woff2') format('woff2');
}"
`;

exports[`css supports keyframe 1`] = `
"@keyframes anim1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}"
`;

exports[`css supports media queries 1`] = `
"@media (max-width: 400px) {
  .red-0-0 {
    color: red;
  }
}"
`;
